tests/components/backup/test_websocket.py

```diff
@@ -115,29 +116,30 @@ async def test_remove(
 
 
 @pytest.mark.parametrize(
-    "with_hassio",
+    ("with_hassio", "number_of_messages"),
     [
-        pytest.param(True, id="with_hassio"),
-        pytest.param(False, id="without_hassio"),
+        pytest.param(True, 1, id="with_hassio"),
+        pytest.param(False, 2, id="without_hassio"),
     ],
 )
+@pytest.mark.usefixtures("mock_backup_generation")
 async def test_generate(
     hass: HomeAssistant,
     hass_ws_client: WebSocketGenerator,
+    freezer: FrozenDateTimeFactory,
     snapshot: SnapshotAssertion,
     with_hassio: bool,
+    number_of_messages: int,
 ) -> None:
     """Test generating a backup."""
     await setup_backup_integration(hass, with_hassio=with_hassio)
 
     client = await hass_ws_client(hass)
+    freezer.move_to("2024-11-13 12:01:00+01:00")
     await hass.async_block_till_done()
 
-    with patch(
-        "homeassistant.components.backup.manager.BackupManager.async_create_backup",
-        return_value=TEST_BACKUP,
-    ):
```

Comment:
The patching is replaced by mocking the backup file creation to better test the cooperation between the WS handler and the backup manager